### PR TITLE
fix: Change the way we update the quality selector item to solve an issues with VideoJS 7...

### PR DESCRIPTION
**Pull request by mcarriere**
...where the menu bar would not fadeout after the quality selector was used.
Also updated all the libs in package.json to the latest available.

### CODE REVIEW

**Comment by mcarriere** [comment]
Tested with VideoJS 7 and 6

**Comment by mcarriere** [comment]
sure, no problem! I'll split everything in multiple commits and fit my commit messages

**Comment by mcarriere** [comment]
hopefully, my 3 commits are now compliant with your commitlint rules. I also added a more detailed explanation to all the commits onto what I changed and why.

File: package.json [hunk 1]
```diff
@@ -25,32 +25,33 @@
...
+    "grunt-contrib-uglify": "4.0.1",
     "grunt-contrib-watch": "1.1.0",
-    "grunt-eslint": "20.0.0",
-    "grunt-postcss": "0.8.0",
-    "grunt-sass": "2.0.0",
-    "grunt-sass-lint": "0.2.2",
+    "grunt-eslint": "21.0.0",
+    "grunt-postcss": "0.9.0",
+    "grunt-sass": "3.0.2",
+    "node-sass": "^4.12.0",
```
[the remark this answers is not in the document]
Comment (mcarriere) [reply package.json]: fixed

File: package.json [hunk 2]
```diff
@@ -25,32 +25,33 @@
...
     "@silvermine/sass-lint-config": "1.1.0",
-    "autoprefixer": "7.1.1",
-    "coveralls": "3.0.2",
-    "eslint": "4.0.0",
+    "autoprefixer": "9.5.1",
+    "coveralls": "3.0.3",
+    "eslint": "5.16.0",
     "expect.js": "0.3.1",
-    "grunt": "1.0.3",
+    "grunt": "^1.0.4",
```
[the remark this answers is not in the document]
Comment (mcarriere) [reply package.json]: fixed

**Comment by mcarriere** [comment]
> @mcarriere Thank you for the PR! Nice job. I have just a couple of feedback items for you in this review. Also, could you update the commit message subject on this commit: [b17858a](https://github.com/silvermine/videojs-quality-selector/commit/b17858a6d38d57ea5fd4ea7cfc353e4f7d7df0a9) to be 72 characters or less?
> 
> Thanks!
commit: [b17858a] subject has been fixed, apparently I can't count :/

File: src/js/components/QualitySelector.js [hunk 3]
```diff
@@ -50,17 +48,16 @@ module.exports = function(videojs) {
...
+       * @param source string player source currently selected
        */
-      setSelectedSource: function(source) {
-         var src = (source ? source.src : undefined);
-
-         if (this.selectedSrc !== src) {
-            this.selectedSrc = src;
-            this.update();
-         }
+      unsetPreviousSource: function(source) {
```
[the remark this answers is not in the document]
Comment (mcarriere) [reply src/js/components/QualitySelector.js]: you are right, I got lost for quite some time in that update() rabbit hole that I forgot to re-evaluate if my changes could be simplified once I was done. I updated my commit with only a few lines changed in setSelectedSource()

**Comment by mcarriere** [comment]
thanks for accepting this :) Would it be possible to create a new release?

**Comment by yokuze** [comment]
Yes, we should do that soon. However, I'd like to fix #31 first.

**Comment by jthomerson** [comment]
@mcarriere this is available in `@silvermine/videojs-quality-selector@1.2.1`

**Comment by mcarriere** [comment]
> @mcarriere this is available in `@silvermine/videojs-quality-selector@1.2.1`
Thanks a lot!
Do you know why it's not on https://www.npmjs.com/package/silvermine-videojs-quality-selector ?

**Comment by yokuze** [comment]
@mcarriere That's the old version of the package. We moved the package to the `@silvermine` org a while back: https://www.npmjs.com/package/@silvermine/videojs-quality-selector

**Comment by mcarriere** [comment]
ok thx!

**Pull request merged**